torchaudio/models/_wavernn.py

```diff
@@ -99,9 +99,129 @@ def forward(self, x: Tensor) -> Tensor:
+    It is a block used in WaveRNN. WaveRNN is based on the paper "Efficient Neural Audio Synthesis".
+    Nal Kalchbrenner, Erich Elsen, Karen Simonyan, Seb Noury, Norman Casagrande, Edward Lockhart,
+    Florian Stimberg, Aaron van den Oord, Sander Dieleman, Koray Kavukcuoglu. arXiv:1802.08435, 2018.
+
+    Args:
+        upsample_scales: the list of upsample scales (required).
+        res_blocks: the number of ResBlock in stack (default=10).
+        input_dims: the number of input sequence (default=100).
+        hidden_dims: the number of compute dimensions (default=128).
+        output_dims: the number of output sequence (default=128).
+        pad: the number of kernal size (pad * 2 + 1) in the first Conv1d layer (default=2).
+
+    Examples::
+        >>> upsamplenetwork = _UpsampleNetwork(upsample_scales=[4, 4, 16],
+                                               res_blocks=10,
+                                               input_dims=100,
+                                               hidden_dims=128,
+                                               output_dims=128,
+                                               pad=2)
+        >>> input = torch.rand(10, 100, 512)
+        >>> output = upsamplenetwork(input)
+    """
+
+    def __init__(self,
+                 upsample_scales: List[int],
+                 res_blocks: int = 10,
+                 input_dims: int = 100,
+                 hidden_dims: int = 128,
+                 output_dims: int = 128,
+                 pad: int = 2) -> None:
+        super().__init__()
+
+        total_scale = 1
+        for upsample_scale in upsample_scales:
+            total_scale *= upsample_scale
+
+        self.indent = pad * total_scale
+        self.resnet = _MelResNet(res_blocks, input_dims, hidden_dims, output_dims, pad)
+        self.resnet_stretch = _Stretch2d(total_scale, 1)
+
+        up_layers = []
+        for scale in upsample_scales:
+            k_size = (1, scale * 2 + 1)
+            padding = (0, scale)
+            stretch = _Stretch2d(scale, 1)
+            conv = nn.Conv2d(in_channels=1, out_channels=1, kernel_size=k_size, padding=padding, bias=False)
+            conv.weight.data.fill_(1. / k_size[1])
+            up_layers.append(stretch)
+            up_layers.append(conv)
+        self.upsample_layers = nn.Sequential(*up_layers)
+
+    def forward(self, x: Tensor) -> Tensor:
+        r"""Pass the input through the _UpsampleNetwork layer.
+
+        Args:
+            x: the input sequence to the _UpsampleNetwork layer (required).
+
+        Shape:
+            - x: :math:`(N, S, T)`.
+            - output: :math:`(N, (T - 2 * pad) * Total_Scale, S)`, `(N, (T - 2 * pad) * total_scale, P)`.
+        where N is the batch size, S is the number of input sequence, T is the length of input sequence.
+        P is the number of output sequence. Total_Scale is the product of all elements in upsample_scales.
```

Comment:
nit: `P = output_dims` ? just do that, or specify that `P = output_dims`.

Reply:
This name has been updated. I use `n_output` here to match other places. No single letter is used in docstring now.